features/dd-sdk-android-session-replay/src/main/kotlin/com/datadog/android/sessionreplay/internal/utils/DrawableUtils.kt

```diff
@@ -13,22 +13,67 @@ import android.util.DisplayMetrics
 import androidx.annotation.MainThread
 import com.datadog.android.sessionreplay.internal.recorder.wrappers.BitmapWrapper
 import com.datadog.android.sessionreplay.internal.recorder.wrappers.CanvasWrapper
+import kotlin.math.sqrt
 
 internal class DrawableUtils(
     private val bitmapWrapper: BitmapWrapper = BitmapWrapper(),
     private val canvasWrapper: CanvasWrapper = CanvasWrapper()
 ) {
+    /**
+     * This method attempts to create a bitmap from a drawable, such that the bitmap file size will
+     * be equal or less than a given size. It does so by modifying the dimensions of the
+     * bitmap, since the file size of a bitmap can be known by the formula width*height*color depth
+     */
     @MainThread
     @Suppress("ReturnCount")
-    internal fun createBitmapFromDrawable(drawable: Drawable, displayMetrics: DisplayMetrics): Bitmap? {
-        val bitmapWidth = if (drawable.intrinsicWidth <= 0) 1 else drawable.intrinsicWidth
-        val bitmapHeight = if (drawable.intrinsicHeight <= 0) 1 else drawable.intrinsicHeight
-        val bitmap = bitmapWrapper.createBitmap(displayMetrics, bitmapWidth, bitmapHeight, Bitmap.Config.ARGB_8888)
+    internal fun createBitmapOfApproxSizeFromDrawable(
+        drawable: Drawable,
+        displayMetrics: DisplayMetrics,
+        requestedSizeInBytes: Int = DESIRED_BITMAP_SIZE_IN_BYTES
+    ): Bitmap? {
+        val dimensions = getScaledWidthAndHeight(drawable, requestedSizeInBytes)
+
+        val bitmap = bitmapWrapper.createBitmap(
+            displayMetrics,
+            dimensions.first,
+            dimensions.second,
+            Bitmap.Config.ARGB_8888
+        )
             ?: return null
 
         val canvas = canvasWrapper.createCanvas(bitmap) ?: return null
         drawable.setBounds(0, 0, canvas.width, canvas.height)
         drawable.draw(canvas)
```

Comment:
so how does writing the scaled bitmap works? I'm asking, because it seems we don't specify any transformation matrix. Say we have `BitmapDrawable` and its bitmap is 1000x1000, will it be properly recorded to the bitmap of size 100x100?

Reply:
It seems to me that there's is automatic scaling applied. `Drawable.draw` is calling `drawBitmap`, and the comment for this method says: "Draw the specified bitmap, scaling/translating automatically to fill the destination rectangle."

Reply:
I'm wondering if scaling also applies to other draw operation (`drawRect`, `drawCircle`, etc.) which take coordinates as parameters.

Reply:
This can be tested inside the sample app by adding these particular cases.

Reply:
I've added some shapes to the sample app